Multisig/UI/Assets/AssetsViewModel.swift

```diff
@@ -8,17 +8,37 @@
 
 import Foundation
 import Combine
+import SwiftCryptoTokenFormatter
+import BigInt
 
-struct TokenBalance: Identifiable {
-    var id: String {
-        address
+struct TokenBalance: Identifiable, Hashable {
+    var id: Int {
+        hashValue
     }
-    var imageURL: URL {
-        URL(string: "https://gnosis-safe-token-logos.s3.amazonaws.com/\(address).png")!
+    var imageURL: URL? {
+        // will be replaced when https://github.com/gnosis/safe-transaction-service/issues/86 is ready
+        guard let address = address else { return nil }
+        return URL(string: "https://gnosis-safe-token-logos.s3.amazonaws.com/\(String(describing: address)).png")!
     }
-    let address: String
+    let address: String?
+    let symbol: String
     let balance: String
     let balanceUsd: String
+
+    init(_ response: SafeBalancesRequest.Response) {
+        self.address = response.tokenAddress
+        self.symbol = response.token?.symbol ?? "ETH"
+
+        let tokenFormatter = TokenFormatter()
```

Comment:
I think both tokenFormatter and currencyFormatter can be defined as shared constants on app level, no need to have an instance foreach token

Reply:
In different parts of the app we might need to use different formatting options. I think it is ok to have it like this.